### PR TITLE
added getSubscriptions and countSubscriptions

### PR DESCRIPTION
**Pull request by firatoltulu**
sometimes we need  how many topic in messages 

i added two methods in pubsubs

getSubscriptions ,
countSubscriptions

### CODE REVIEW

**Comment by mroderick** [comment]
Thank you for your PR 👍 

Would you mind adding some unit tests for the two new methods?

**Comment by mroderick** [comment]
Oh, and fix the linting violation to make the linter happy :)

**Comment by mroderick** [comment]
This looks like it is fixing #146

**Comment by firatoltulu** [comment]
ı updated unit test and readme.

File: docs/docs.md [hunk 1]
```diff
@@ -108,21 +113,29 @@ Clears all subscriptions
...
 
-- When passed a topic, removes all subscriptions for that topic (hierarchy)
+## subscribeOnce(value)
+Removes subscriptions

- When passed a token, removes a specific subscription.

- When passed a function, removes all subscriptions for that function

- When passed a topic, removes all subscriptions for that topic (hierarchy)
```
Comment (mroderick) [inline docs/docs.md]: Some control characters have snuck into the README

```suggestion
Removes subscriptions - When passed a token, removes a specific subscription - When passed a function, removes all subscriptions for that function - When passed a topic, removes all subscriptions for that topic (hierarchy)
```

File: docs/docs.md [hunk 2]
```diff
@@ -133,21 +146,17 @@ Removes subscriptions
 
 **Example**  
 ```js
-// Unsubscribing with a token
-var token = PubSub.subscribe('mytopic', myFunc);
-PubSub.unsubscribe(token);
+// Unsubscribing with a tokenvar token = PubSub.subscribe('mytopic', myFunc);PubSub.unsubscribe(token);
```
Comment (mroderick) [inline docs/docs.md]: Control characters

```suggestion
// Unsubscribing with a token var token = PubSub.subscribe('mytopic', myFunc); PubSub.unsubscribe(token);
```

File: docs/docs.md [hunk 3]
```diff
@@ -133,21 +146,17 @@ Removes subscriptions
...
+// Unsubscribing with a token
var token = PubSub.subscribe('mytopic', myFunc);
PubSub.unsubscribe(token);
 ```
 **Example**  
 ```js
-// Unsubscribing with a function
-PubSub.unsubscribe(myFunc);
+// Unsubscribing with a function
PubSub.unsubscribe(myFunc);
```
Comment (mroderick) [inline docs/docs.md]: Control characters

```suggestion
// Unsubscribing with a function PubSub.unsubscribe(myFunc);
```

File: docs/docs.md [hunk 4]
```diff
@@ -133,21 +146,17 @@ Removes subscriptions
...
 ```
 **Example**  
 ```js
-// Unsubscribing from a topic
-PubSub.unsubscribe('mytopic');
+// Unsubscribing from a topic
PubSub.unsubscribe('mytopic');
```
Comment (mroderick) [inline docs/docs.md]: Control characters

```suggestion
// Unsubscribing from a topic PubSub.unsubscribe('mytopic');
```

File: src/pubsub.js [hunk 5]
```diff
@@ -229,6 +230,44 @@
...
     };
 
+    /** 
+       Count subscriptions by the topic
+     * @function
+     * @public
+     * @alias countSubscriptions
+     * @return { Array }
+    */
+
```
Comment (mroderick) [inline src/pubsub.js]: Remove the generous white space

```suggestion
```

File: src/pubsub.js [hunk 6]
```diff
@@ -229,6 +230,44 @@
...
+        var count = 0;
+        for (m in messages){
+            if (messages.hasOwnProperty(m) && m.indexOf(topic) === 0){
+                count++
+            }
+        }
+        return count;
+    };
+
+
```
Comment (mroderick) [inline src/pubsub.js]: Remove the generous white space

```suggestion
```

File: test/test-getSubscriptions.js [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+        var topic = TestHelper.getUniqueString(),
+            spy1 = sinon.spy();
+
+        PubSub.subscribe(topic, spy1);
+
+        var subscriptions = PubSub.getSubscriptions(topic).length;
+
+        assert.equals(subscriptions,1);
+    });
+
```
Comment (mroderick) [inline test/test-getSubscriptions.js]: Remove the generous white space

```suggestion
```

File: test/test-getSubscriptions.js [hunk 8]
```diff
@@ -0,0 +1,22 @@
+'use strict';
+
+var PubSub = require('../src/pubsub'),
+    TestHelper = require('./helper'),
+    refute = require('referee').refute,
+    assert = require('referee').assert,
+    sinon = require('sinon');
+
+describe('getSubscriptions method', function () {
+
```
Comment (mroderick) [inline test/test-getSubscriptions.js]: Remove the generous white space

```suggestion
```

File: test/test-countSubscriptions.js [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+        var topic = TestHelper.getUniqueString(),
+            spy1 = sinon.spy();
+
+        PubSub.subscribe(topic, spy1);
+
+        var counts = PubSub.countSubscriptions(topic);
+
+        assert.equals(counts,1);
+    });
+
```
Comment (mroderick) [inline test/test-countSubscriptions.js]: Remove the generous white space

```suggestion
```

File: test/test-countSubscriptions.js [hunk 10]
```diff
@@ -0,0 +1,22 @@
+'use strict';
+
+var PubSub = require('../src/pubsub'),
+    TestHelper = require('./helper'),
+    refute = require('referee').refute,
+    assert = require('referee').assert,
+    sinon = require('sinon');
+
+describe('test-countSubscriptions method', function () {
+
```
Comment (mroderick) [inline test/test-countSubscriptions.js]: Remove the generous white space

```suggestion
```

File: test/test-countSubscriptions.js [hunk 11]
```diff
@@ -0,0 +1,22 @@
+'use strict';
+
+var PubSub = require('../src/pubsub'),
+    TestHelper = require('./helper'),
+    refute = require('referee').refute,
```
Comment (mroderick) [inline test/test-countSubscriptions.js]: `refute` is not in use in this file

```suggestion
```

File: test/test-getSubscriptions.js [hunk 12]
```diff
@@ -0,0 +1,22 @@
+'use strict';
+
+var PubSub = require('../src/pubsub'),
+    TestHelper = require('./helper'),
+    refute = require('referee').refute,
```
Comment (mroderick) [inline test/test-getSubscriptions.js]: `refute` is not in use in this file

```suggestion
```

File: src/pubsub.js [hunk 13]
```diff
@@ -229,6 +230,44 @@
...
+        var list = [];
+        for (m in messages){
+            if (messages.hasOwnProperty(m) && m.indexOf(topic) === 0){
+                list.push(m);
+            }
+        }
+        return list;
+    };
+
+
```
Comment (mroderick) [inline src/pubsub.js]: Remove the generous white space
```suggestion
```

**Comment by firatoltulu** [comment]
Thank you for information and changes.

**Comment by firatoltulu** [comment]
Can you merge this?

**Comment by mroderick** [comment]
This has been published in `v1.8.0`

**Pull request merged**